pkg/store/store.go

```diff
@@ -0,0 +1,247 @@
+
+	return nil
+}
+
+// GetContent returns just the content bytes for the given path
+func (s *store) GetContent(path string) ([]byte, bool) {
+	s.mu.RLock()
+	defer s.mu.RUnlock()
+
+	file, exists := s.files[path]
+	if !exists {
+		return nil, false
+	}
+
+	// Return a copy to avoid external modifications
+	return append([]byte(nil), file.Content...), true
+}
+
+// RotationEvents returns a read-only channel for rotation events
+func (s *store) RotationEvents() <-chan RotationEvent {
+	return s.rotationCh
+}
+
+// Errors returns a read-only channel for errors
+func (s *store) Errors() <-chan error {
+	return s.errorCh
+}
+
+// startWatching starts a goroutine that handles filesystem events
+func (s *store) startWatching() {
+	go func() {
+		s.logger.Info("Starting filesystem watcher")
+
+		// Ensure cleanup when goroutine exits
+		defer func() {
+			s.logger.Info("Cleaning up filesystem watcher resources")
+			close(s.rotationCh)
+			close(s.errorCh)
+			s.watcher.Close()
+		}()
+
+		for {
+			select {
+			case <-s.ctx.Done():
+				s.logger.Info("Stopping filesystem watcher due to context cancellation")
+				return
+			case event, ok := <-s.watcher.Events:
+				if !ok {
+					s.logger.Info("Filesystem watcher events channel closed")
+					return
+				}
+				s.handleFileEvent(event)
+			case err, ok := <-s.watcher.Errors:
+				if !ok {
+					s.logger.Info("Filesystem watcher errors channel closed")
+					return
+				}
+				s.logger.Error(err, "Filesystem watcher error")
+
+				// Send error to error channel
+				select {
+				case s.errorCh <- err:
+				default:
+					s.logger.Info("Error channel full, dropping error")
+				}
+			}
+		}
+	}()
+}
+
+// handleFileEvent processes a filesystem event
+func (s *store) handleFileEvent(event fsnotify.Event) {
+	s.mu.Lock()
+	defer s.mu.Unlock()
+
+	path := event.Name
+
+	// Check if this file is being tracked
+	file, exists := s.files[path]
+	if !exists {
+		return
+	}
+
+	s.logger.Info("File event received", "path", path, "op", event.Op.String())
+
+	// Handle different event types
+	if event.Has(fsnotify.Remove) || event.Has(fsnotify.Rename) {
+		// File was deleted or renamed - return error
+		err := fmt.Errorf("file was deleted or renamed: %s", path)
+		s.logger.Error(err, "File no longer exists")
+
+		// Send error to error channel
+		select {
+		case s.errorCh <- err:
+		default:
+			s.logger.Info("Error channel full, dropping error")
+		}
+		return
+	}
+
+	if event.Has(fsnotify.Write) || event.Has(fsnotify.Create) {
+		// File was modified or created, reload content
+		if err := s.refreshFileContent(path, file); err != nil {
+			s.logger.Error(err, "Failed to refresh file content", "path", path)
+
+			// Send error to error channel
+			select {
+			case s.errorCh <- err:
+			default:
+				s.logger.Info("Error channel full, dropping error")
+			}
+		} else {
+			// Send rotation event for file update
+			select {
+			case s.rotationCh <- RotationEvent{Path: path}:
```

Comment:
where do we leverage rotation events? or will they eventually be emitted to other resources?

Reply:
They'll be used to requeue managed domain controller. Will be needed

Reply:
Error one is for health check to ensure everything is healthy